wagtail/models/__init__.py

```diff
@@ -1299,6 +1299,164 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.title
 
+    def _process_sibling(self, nodes, parent):
+        """
+        Create child nodes according to the node_order_by attribute.
+        """
+
+        newpositions = []
+        siblings_so_far = self.get_siblings()
+        for i, node in enumerate(nodes):
+            pos = self._prepare_pos_var_for_add_sibling("sorted-sibling")
+            node.depth = self.depth
+            siblings = self.get_sorted_pos_queryset(siblings_so_far, node)
+            try:
+                newpos = siblings.all()[0]._get_lastpos_in_path()
+            except IndexError:
+                newpos = None
+                pos = "last-sibling"
+            newpositions.append(newpos)
+            _, newpath = self.reorder_nodes_before_add_or_move(
+                pos, newpos, self.depth, self, siblings, None, False
+            )
+
+            parent.numchild += 1
+            node.path = newpath
+            # Add node to siblings_so_far. Make node a queryset to union with siblings_so_far
+            siblings_so_far = siblings_so_far.union(siblings)
+
+        return nodes
+
+    def _process_leaf(self, nodes):
+        """
+        Process children to add to a leaf node
+        """
+        # Get number of children
+
+        step = 1
+        max_length = self.__class__._meta.get_field("path").max_length
+        for node in nodes:
+            node.depth = self.depth + 1
+            node.path = self.__class__._get_path(self.path, node.depth, step)
+            step += 1
+            if len(node.path) > max_length:
+                raise ValidationError(
+                    "The new node is too deep in the tree, try \
+                      increasing the path.max_length property \
+                    and UPDATE your database"
+                )
+            self.numchild += 1
+            node._cached_parent_obj = self
+
+        self.save()
```

Comment:
I think that's a great suggestion. It'll help separate the preparation step and the saving step. Since we're loading the children to the db and saving changes to the parent page I think we can keep them in a separate method.